detectable/src/main/java/com/synopsys/integration/detectable/detectables/maven/cli/MavenCliExtractor.java

```diff
@@ -46,13 +49,27 @@ public Extraction extract(File directory, ExecutableTarget mavenExe, MavenCliExt
         List<String> includedScopes = mavenCliExtractorOptions.getMavenIncludedScopes();
         List<String> excludedModules = mavenCliExtractorOptions.getMavenExcludedModules();
         List<String> includedModules = mavenCliExtractorOptions.getMavenIncludedModules();
+        Boolean includeShadedDependencies = mavenCliExtractorOptions.getMavenIncludeShadedDependencies();
+
+        if(includeShadedDependencies) {
+            try {
+                mavenProjectInspectorDetectable.setIncludeShadedDependencies(true);
+                mavenProjectInspectorDetectable.extractable();
+                Extraction extraction = mavenProjectInspectorDetectable.extract(extractionEnvironment);
+                shadedDependencies = mavenProjectInspectorDetectable.getShadedDependencies();
+            } catch (Exception e) {
+               throw new RuntimeException("There was an error extracting the shaded dependencies from Project Inspector. Please ensure that you have the latest version of Project Inspector 2024.2.0",e);
```

Comment:
This seems risky to mention the latest at the time of writing. Feels like that could change and we forget to update this. Maybe just pull the specific version or get thoughts from Chris?

Reply:
Good suggestion, I will work with Chris to find a nice way to frame this. Thanks.